build/build-in-docker

```diff
@@ -24,30 +24,30 @@ set -e
 SCRIPTDIR=$(cd $(dirname $0); pwd)
 
 LOCAL_MAVEN_REPO=${LOCAL_MAVEN_REPO:-"$HOME/.m2/repository"}
-CUDF_USE_PER_THREAD_DEFAULT_STREAM=${CUDF_USE_PER_THREAD_DEFAULT_STREAM:-ON}
 USE_GDS=${USE_GDS:-ON}
 export CMAKE_GENERATOR=${CMAKE_GENERATOR:-"Ninja"}
+export DOCKER_BUILD_EXTRA_ARGS="--build-arg TARGETPLATFORM=linux/amd64 --build-arg CMAKE_ARCH=x86_64"
+export CUDA_VERSION=${CUDA_VERSION:-11.8.0}
+CUDA_CLASSIFIER=cuda${CUDA_VERSION%%.*}
+USE_SANITIZER=${USE_SANITIZER:-ON}
+BUILD_FAULTINJ=${BUILD_FAULTINJ:-ON}
 
 if (( $# == 0 )); then
   echo "Usage: $0 <Maven build arguments>"
   exit 1
 fi
 
-_CUDF_CLEAN_SKIP=""
-# if ccache is enabled and libcudf.clean.skip not provided
-# by the user remove the cpp build directory
-#
-if [[ "$CCACHE_DISABLE" != "1" ]]; then
-  if [[ ! "$*" =~ " -Dlibcudf.clean.skip=" ]]; then
-    # Don't skip clean if ccache is enabled
-    # unless the user overrides
-    _CUDF_CLEAN_SKIP="-Dlibcudf.clean.skip=false"
-  fi
+# Set env for arm64 build, The possible values of 'uname -m' : [x86_64/i386/aarch64/mips/...]
+if [ "$(uname -m)" == "aarch64" ]; then
+  USE_GDS="OFF" # The GDS cuFiles RDMA libraries are not included in the arm64 CUDA toolkit.
+  USE_SANITIZER="ON" # OOM failures in rmm occur during arm64 unit tests when USE_SANITIZER=OFF
+  BUILD_FAULTINJ="OFF" # libcupti_static.a linked by cufaultinj, does not exist in the arm64 CUDA toolkit.
+  export DOCKER_BUILD_EXTRA_ARGS="--build-arg TARGETPLATFORM=linux/arm64 --build-arg CMAKE_ARCH=aarch64"
```

Comment:
"Besides DOCKER_BUILD_EXTRA_ARGS, all other parameters/variables defined in build-in-docke are for the use of run-in-docker."

"Do you mean that we should merge these two files, build-in-docke and run-in-docker, into one file called run-in-docker?"

Reply:
No, definitely do not merge these two scripts.  build-in-docker is designed for building, it knows about maven, etc.  run-in-docker is a lower-level script used by build-in-docker that developers can use to run something _other_ than maven.  What I'm saying here is that this line looks like it's trying to force a certain image to use based on the platform architecture.  That's something run-in-docker should do, not this script, since it has nothing to do with just building with maven -- anything we want to run needs to pick the proper image, thus it's a run-in-docker change.

To be honest, I was surprised to see so many changes to build-in-docker since run-in-docker is the script that's supposed to know the intimate details of what image to pick.  Given cupti is not available on arm64, it makes some sense, but a lot of other changes don't.

Reply:
agree, let me update it!